scheduling/first_come_first_served.py

```diff
@@ -1,66 +1,83 @@
 # Implementation of First Come First Served CPU scheduling Algorithm
 # In this Algorithm we just care about the order that the processes arrived
 # without carring about their duration time
-
-# Function to calculate the waiting time of each process
-
-
-def findWaitingTime(processes, n, duration_time, waiting_time):
-
-    # Initialising the first prosecces' waiting time with 0
-    waiting_time[0] = 0
-
-    # calculating waiting time
-    for i in range(1, n):
+# https://en.wikipedia.org/wiki/Scheduling_(computing)#First_come,_first_served
+from typing import List
+from functools import reduce
+
+
+def calculate_waiting_time(
```

Comment:
Let’s remove `number_of_processes` because that is identical to `len(duration_time)`.  Do the same for other functions below.